### PR TITLE
v.colors: Fix resource leak issue in scan_attr.c file

### PR DESCRIPTION
**Pull request by ShubhamDesai**
This pull request addresses resource leak issue identified by coverity scan (CID : 1207685)
Used existing function Vect_destroy_field_info() to fix the memory leak issue.

### CODE REVIEW

**Comment by echoix** [comment]
@nilason What do we (I) have to check with these kind of PRs? When does it apply, when should it NOT be used? What should I look for of "missing" changes that should also be part of a PR? Are there other gotchas to be aware of?

I'm also trying to learn this, there's been a couple of similar PRs recently, and didn't have a clue on what to look for. Every info/knowledge that I'd get I can use for future PRs when I do my screening (when I can), so I can weed out the easy issues and keep you and other for more advanced/specialized questions.

File: vector/v.colors/scan_attr.c [hunk 1]
```diff
@@ -42,6 +42,7 @@ int scan_attr(struct Map_info *Map, int layer, const char *column_name,
                                  &cvarr);
     if (nrec < 1) {
         G_important_message(_("No data selected"));
+        Vect_destroy_field_info(fi);
```
Comment (nilason) [inline vector/v.colors/scan_attr.c:45]: Here you can also close the database (`driver`) before return.

**Review by nilason — CHANGES_REQUESTED** [review]

File: vector/v.colors/scan_attr.c (hunk 1 above)
Comment (ShubhamDesai) [reply vector/v.colors/scan_attr.c:45]: done

**Comment by nilason** [comment]
> @nilason What do we (I) have to check with these kind of PRs? When does it apply, when should it NOT be used? What should I look for of "missing" changes that should also be part of a PR? Are there other gotchas to be aware of?
> 
> I'm also trying to learn this, there's been a couple of similar PRs recently, and didn't have a clue on what to look for. Every info/knowledge that I'd get I can use for future PRs when I do my screening (when I can), so I can weed out the easy issues and keep you and other for more advanced/specialized questions.

It’s terribly difficult to give you a general advice. Sorry :)

**Review by nilason — APPROVED** [review]

**Pull request merged**